dual_screen/android/src/main/kotlin/com/builttoroam/dual_screen/DualScreenPlugin.kt

```diff
@@ -0,0 +1,132 @@
+package com.builttoroam.dual_screen
+
+import android.app.Activity
+import android.content.Context
+import android.view.View
+import android.view.View.OnLayoutChangeListener
+import androidx.annotation.NonNull
+import com.microsoft.device.display.DisplayMask
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.EventChannel
+import io.flutter.plugin.common.EventChannel.EventSink
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+/** DualScreenPlugin */
+public class DualScreenPlugin :
+    FlutterPlugin,
+    MethodCallHandler,
+    EventChannel.StreamHandler,
+    ActivityAware {
+    private val METHOD_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/methods"
+    private val EVENT_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/events"
+    private lateinit var methodChannel: MethodChannel
+    private lateinit var eventChannel: EventChannel
+    private var context: Context? = null
+    private var activity: Activity? = null
+
+    fun registerPlugin(context: Context?, messenger: BinaryMessenger?) {
+        activity = this.activity
+        this.context = context
+        MethodChannel(messenger, METHOD_CHANNEL_NAME).setMethodCallHandler(this)
+        EventChannel(messenger, EVENT_CHANNEL_NAME).setStreamHandler(this)
+    }
+
+    override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        this.registerPlugin(binding.getApplicationContext(), binding.getBinaryMessenger())
+    }
+
+    companion object {
+        @JvmStatic
+        fun registerWith(registrar: Registrar) {
+            DualScreenPlugin().registerPlugin(registrar.context(), registrar.messenger())
+        }
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "isDualScreenDevice" -> {
+                result.success(isDualScreenDevice())
+            }
+            "isAppSpanned" -> {
+                if (isDualScreenDevice()) {
```

Comment:
I would usually tend to agree with this approach if we were returning a value for each possibility, however since onMethodCall is a void function i dont think we should be returning a value here

Reply:
Oh, my bad. You're right. I don't know why I assumed we're returning stuff here.